library/src/test/java/com/novoda/downloadmanager/lib/DownloadStatusTest.java

```diff
@@ -0,0 +1,127 @@
+package com.novoda.downloadmanager.lib;
+
+import android.app.DownloadManager;
+
+import org.junit.Test;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+
+public class DownloadStatusTest {
+
+    @Test
+    public void whenVisibilityIsCompletedThenNotificationIsToBeDisplayed() {
+        boolean toBeDisplayed = DownloadStatus.isNotificationToBeDisplayed(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED);
+        assertThat(toBeDisplayed).isTrue();
+    }
+
+    @Test
+    public void whenVisibilityIsOnlyCompletionThenNotificationIsToBeDisplayed() {
+        boolean toBeDisplayed = DownloadStatus.isNotificationToBeDisplayed(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_ONLY_COMPLETION);
+        assertThat(toBeDisplayed).isTrue();
+    }
+
+    @Test
+    public void whenVisibilityIsHiddenThenNotificationIsNotBeDisplayed() {
+        boolean toBeDisplayed = DownloadStatus.isNotificationToBeDisplayed(DownloadManager.Request.VISIBILITY_HIDDEN);
+        assertThat(toBeDisplayed).isFalse();
+    }
+
+    @Test
+    public void whenStatusIs490ThenStatusIsCancelled() {
+        boolean isCancelled = DownloadStatus.isCancelled(490);
+        assertThat(isCancelled).isTrue();
+    }
+
+    @Test
+    public void whenStatusIs4xxThenStatusIsClientError() {
```

Comment:
Yup same as @Electryc https://github.com/novoda/download-manager/pull/83#discussion_r34233926

The test should validate all 4xx values, 450 is validated but 440 is not, that's why of the loop, to make sure that all 4xx values pass the tests

The two extra assertions are there in order to tests the limits, but an isolated test for 399 testing that is not a client error or a 500 or any other range doesn't really add any value because doesn't assert that any value non 4xx is not a client error. That's why I've included the limits in the same test as they are in context.